tests/test_bgp_neighbor.rb

```diff
@@ -388,11 +528,22 @@ def test_bgpneighbor_set_get_transport_passive_only
     end
   end
 
-  def test_bgpneighbor_set_get_update_source
+  def test_set_get_update_source
     %w(default test_vrf).each do |vrf|
-      neighbor = RouterBgpNeighbor.new(@@asn, vrf, @@addr)
-      interfaces = ['loopback1', 'Ethernet1/1', 'ethernet1/1',
-                    neighbor.default_update_source]
+      neighbor = create_neighbor(vrf)
+
+      interfaces = []
+
+      if platform == :ios_xr
+        interfaces = ['loopback1',
+                      'GigabitEthernet0/0/0/0',
+                      'gigabitethernet0/0/0/0',
+                      neighbor.default_update_source]
+      else
+        interfaces = ['loopback1', 'Ethernet1/1', 'ethernet1/1',
+                      neighbor.default_update_source]
```

Comment:
A platform-independent way to do this would be to use the ciscotest 'interfaces' API:

```
my_interfaces = ['loopback1', interfaces[0], interfaces[0].downcase, neighbor.default_update_source]
```